### PR TITLE
update golang to in-support versions (1.16 and 1.17)

### PR DESCRIPTION
**Pull request by joshspicer**
closes https://github.com/microsoft/vscode-dev-containers/issues/1000

Updates our versions of `go` to track latest supported

### TODO
- [ ] Test with `dev` tag
- [ ] Bump repo version to do full release
- [ ] Update [tracking issue](https://github.com/microsoft/vscode-dev-containers/issues/532)
- [ ] Pull into Codespaces

### CODE REVIEW

**Comment by joshspicer** [comment]
builds fine in remote containers 👍 
image

File: containers/go/definition-manifest.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
-	"variants": ["1.16", "1.15"],
-	"definitionVersion": "0.203.0",
+	"variants": ["1.16", "1.17"],
```
Comment (Chuxel) [inline containers/go/definition-manifest.json:2]:
```suggestion
	"variants": ["1.17", "1.16"],
```

The order here matters. The zero index is assigned "latest", so put 1.17 there.

**Review by Chuxel — CHANGES_REQUESTED** [review]
Two things.

It looks like there's a bug in the go-debian script this exposed.  It turns out that `go version` emits a version of `go1.17` instead of `go1.17.0` like you'd normally expect.

Therefore this line

```
if [[ "1.16" > "$(go version | grep -oP 'go\K[0-9]+\.[0-9]+\.[0-9]+')" ]]; then
```
needs to make the last section of the semver optional. I think this should do it:
```
if [[ "1.16" > "$(go version | grep -oP 'go\K[0-9]+\.[0-9]+(\.[0-9]+)?')" ]]; then
```
Go 1.17 recommends using "go install" for CLIs instead of "go get", but older versions still need the old syntax. So this detects the actually installed version (since we may have skipped installing go itself) and only uses `go install` for go 1.16+.

The other is a minor tweak inline...

**Comment by joshspicer** [comment]
nice - that looks good
image

**Review by Chuxel — APPROVED** [review]

**Comment by Chuxel** [comment]
@joshspicer Looks good! Feel free to merge when ready!

**Pull request merged**